platform/ios/src/MGLMapView.mm

```diff
@@ -1364,10 +1380,16 @@ - (void)handleRotateGesture:(UIRotationGestureRecognizer *)rotate
             newDegrees = fminf(newDegrees,  30);
             newDegrees = fmaxf(newDegrees, -30);
         }
-
-        _mbglMap->setBearing(newDegrees, mbgl::ScreenCoordinate { centerPoint.x, centerPoint.y });
-
-        [self notifyMapChange:mbgl::MapChangeRegionIsChanging];
+        
+        if ([self.delegate respondsToSelector:@selector(mapView:shouldChangeFromCamera:toCamera:)]
+            && ![self.delegate mapView:self shouldChangeFromCamera:oldCamera toCamera:self.camera])
```

Comment:
`newCamera` equals `oldCamera` because we haven’t called `mbgl::Map::setBearing()` yet. `newCamera` should instead reflect the camera _after_ applying the rotation.